_rules/SC2-4-6-descriptive-headings.md

```diff
@@ -0,0 +1,184 @@
+---
+name: Heading is descriptive
+test_type: atomic
+
+description: |
+   This rule checks that headings describe the topic or purpose of the content.
+   
+success_criterion:
+- 2.4.6 # Headings and labels
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Dagfinn Rømen
+- Geir Sindre Fossøy
+- Carlos Duarte
+---
+
+## Test procedure
+
+### Applicability
+
+This rule applies to any element with the [semantic role](#semantic-role) of heading that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+**Note**: This rule only applies to elements with the [semantic role](#semantic-role) of heading. Thus, it is a partial check for WCAG 2.0 success criterion 2.4.6, which applies to all headings. "Heading" is used in its general sense and includes headlines and other ways to add a heading to different types of content. This includes elements that are not marked up as headings in the code, but still act visually as headings, e.g. by larger and/or bolder text. 
+
+### Expectation
+
+Each target element describes the topic or purpose of its [section of the content](#section-of-content).
```

Comment:
How do you decide what section belongs to what heading? Stuff in the same block, following the reading order until the next heading?

Reply:
> 
> 
> How do you decide what section belongs to what heading? Stuff in the same block, following the reading order until the next heading?

I would say this can be fixed in the "section-of-content" algorithm.

"A heading that marks the beginning of the section of content."

Might be "A heading that marks the beginning of the section and the next heading of a same or higher level that marks the ending of the section."

Reply:
+1 re: the requirement to define the borders of a section that ties together a heading and the content it describes.

Section boundaries are very subjective to business rules attempting to define boundaries by the relationship between a heading, and relevant content doesn't lend itself well to anything more than a representative definition of section boundaries. Any other approach to defining boundaries may inadvertently and adversely affect a business's freedom to define content structures, semantics and hierarchy according to unique vertical industry/specific business rules.

Any attempts to describe section boundaries should consider/include the following points:

1. Deciding if a heading belongs to a section is a very subject business rule.

2. Relying on anything other than structured markup and nested <section> markup to determine where relevant headings exist and whether or not the heading is relevant to the content that follows is either defined by business rules, training authors accordingly, and parsing the content programmatically.

3. We could make a recommendation to observe the publishing best practice using nested <section> elements to delimit section hierarchy and semantics providing a programmatic structure defining what heading belongs with which content.

4. Visual appearance is not relevant to establishing whether or not a heading effectively describes the associated content isn't relevant in hierarchical sectioning. The only way to determine structure/semantics/navigation meets WCAG is by reviewing markup to ensure the appropriate order of headings, establishing a logical hierarchy.

The operative word here is "describe." We can describe, but we cannot put a hard definition of the heading/content relationship from a content architecture perspective without limiting subjectivity.